sigar_interface.go

```diff
@@ -18,6 +18,11 @@ type Sigar interface {
 
 var ErrNotImplemented error = errors.New("Collection not implemented for this operating system")
 
+// Simple Get() that returns an error
+type Getter interface {
+	Get() error
+}
```

Comment:
Interfaces in go just blew my mind:
> To implement an interface in Go, we just need to implement all the methods in the interface.

So you don't have to declare that a type implements an interface, but it can be cast to an interface type if it implements all those methods.  Compilers are magic